### PR TITLE
Avoid scaling offset twice when spatium is changed

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #16006

It turns out the that issue happened because Score::spatiumChanged() was being inadvertently called twice when changing the staff size in a piece. This should fix it.

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/engraving/tests/parts_data/part-all-appendmeasures.mscx [hunk 1]
```diff
@@ -1267,7 +1267,7 @@
                 <Fingering>
                   <linked>
                     </linked>
-                  <offset x="1.06662" y="-4.71271"/>
+                  <offset x="1.05809" y="-4.67501"/>
```
Comment (RomanPudashkin) [inline src/engraving/tests/parts_data/part-all-appendmeasures.mscx:1270]: Why did you change the test data?

Comment (mike-spa) [reply src/engraving/tests/parts_data/part-all-appendmeasures.mscx:1270]: Ok I've double checked to be sure. In all of these tests, the spatium in the master score is slightly different than the spatium in the part, so all of these tests had slightly wrong offsets because of the wrong scaling calculations. Now all the new offset values should be correct.
